cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -1248,39 +1248,38 @@ runQueryLeadershipSchedule (AnyConsensusModeParams cModeParams) network
-             CurrentEpoch -> do
-               let currentEpochStateQuery = QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryCurrentEpochState
-                   currentEpochQuery = QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryEpoch
-               serCurrentEpochState <- executeQuery era cModeParams localNodeConnInfo currentEpochStateQuery
-               curentEpoch <- executeQuery era cModeParams localNodeConnInfo currentEpochQuery
-               firstExceptT ShelleyQueryCmdLeaderShipError $ hoistEither
-                $ eligibleLeaderSlotsConstaints sbe
-                $ currentEpochEligibleLeadershipSlots
-                    sbe
-                    shelleyGenesis
-                    eInfo
-                    pparams
-                    ptclState
-                    poolid
-                    vrkSkey
-                    serCurrentEpochState
-                    curentEpoch
-
-             NextEpoch -> do
-               let currentEpochStateQuery = QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryCurrentEpochState
-                   currentEpochQuery = QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryEpoch
-               tip <- liftIO $ getLocalChainTip localNodeConnInfo
-
-               curentEpoch <- executeQuery era cModeParams localNodeConnInfo currentEpochQuery
-               serCurrentEpochState <- executeQuery era cModeParams localNodeConnInfo currentEpochStateQuery
-
-               firstExceptT ShelleyQueryCmdLeaderShipError $ hoistEither
-                $ eligibleLeaderSlotsConstaints sbe
-                $ nextEpochEligibleLeadershipSlots sbe shelleyGenesis
-                    serCurrentEpochState ptclState poolid vrkSkey pparams
-                    eInfo (tip, curentEpoch)
+      schedule <- case whichSchedule of
+        CurrentEpoch -> do
+          let currentEpochQuery = QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryEpoch
+          serCurrentEpochState <- executeQuery era cModeParams localNodeConnInfo $
+            QueryInEra eInMode $ QueryInShelleyBasedEra sbe (QueryPoolDistr (Just (Set.singleton poolid)))
+          curentEpoch <- executeQuery era cModeParams localNodeConnInfo currentEpochQuery
+          firstExceptT ShelleyQueryCmdLeaderShipError $ hoistEither
+            $ eligibleLeaderSlotsConstaints sbe
+            $ currentEpochEligibleLeadershipSlots
+              sbe
+              shelleyGenesis
+              eInfo
+              pparams
+              ptclState
+              poolid
+              vrkSkey
+              serCurrentEpochState
+              curentEpoch
+
+        NextEpoch -> do
+          let currentEpochStateQuery = QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryCurrentEpochState
+              currentEpochQuery = QueryInEra eInMode $ QueryInShelleyBasedEra sbe QueryEpoch
+          tip <- liftIO $ getLocalChainTip localNodeConnInfo
+
+          curentEpoch <- executeQuery era cModeParams localNodeConnInfo currentEpochQuery
```

Comment:
Out of curiosity, why can't we just query the pool distribution for the next epoch?

Reply:
Can you elaborate?  Do you mean don't query for the current epoch but query the pool distribution instead?

Reply:
I simplified the code so that `curentEpoch` is computed once rather than in each case of `whichSchedule`.

Reply:
Ignore me, I misread the code